### PR TITLE
Vehicle update 9473 02

### PR DESCRIPTION
**Pull request by hdpoliveira**

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -6393,26 +6396,26 @@ int32_t Vehicle::UpdateMotionDodgems()
  *
  *  rct2: 0x006DD365
  */
-bool vehicle_update_dodgems_collision(Vehicle* vehicle, int16_t x, int16_t y, uint16_t* spriteId)
+bool Vehicle::UpdateDodgemsCollision(int16_t curX, int16_t curY, uint16_t* spriteId)
 {
-    uint16_t bp = (vehicle->var_44 * 30) >> 9;
-    uint32_t trackType = vehicle->track_type >> 2;
+    uint16_t bp = (var_44 * 30) >> 9;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:6401]: bp is the dodgems vehicles radius. Could you rename the variable to something like that.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -6393,26 +6396,26 @@ int32_t Vehicle::UpdateMotionDodgems()
  *
  *  rct2: 0x006DD365
  */
-bool vehicle_update_dodgems_collision(Vehicle* vehicle, int16_t x, int16_t y, uint16_t* spriteId)
+bool Vehicle::UpdateDodgemsCollision(int16_t curX, int16_t curY, uint16_t* spriteId)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:6399]: another refactoring project is replacing these x and y variables with the respective structs. In this instance this is a `CoordsXY`. Can you replace x and y with a `CoordsXY`. All calls can construct this with `{x ,y}`

Comment (hdpoliveira) [reply src/openrct2/ride/Vehicle.cpp]: During this I actually noticed this does not Update the vehicle in any way, so I renamed it, and added a const to it.

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -6393,26 +6396,32 @@ int32_t Vehicle::UpdateMotionDodgems()
  *
  *  rct2: 0x006DD365
  */
-bool vehicle_update_dodgems_collision(Vehicle* vehicle, int16_t x, int16_t y, uint16_t* spriteId)
+static bool wouldCollideWithDodgemsTrackEdge(
+    CoordsXY coords, CoordsXY trackLocation, uint32_t trackType, uint16_t dodgemsCarRadius)
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:6400]:
```suggestion
    const CoordsXY& coords, cons CoordsXY& trackLocation, uint32_t trackType, uint16_t dodgemsCarRadius)
```
(formatting will probably need to be revisited as well)

File: src/openrct2/ride/Vehicle.h [hunk 4]
```diff
@@ -305,7 +305,11 @@ struct Vehicle : SpriteBase
     void SetState(VEHICLE_STATUS vehicleStatus, uint8_t subState = 0);
     bool IsGhost() const;
     void UpdateSoundParams(std::vector<rct_vehicle_sound_params>& vehicleSoundParamsList) const;
+    bool DodgemsCarWouldCollideAt(CoordsXY coords, uint16_t* spriteId) const;
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.h:308]:
```suggestion
    bool DodgemsCarWouldCollideAt(const CoordsXY& coords, uint16_t* spriteId) const;
```

**Pull request merged**